src/Views/OperatingSystemView.vala

```diff
@@ -713,4 +731,113 @@ public class About.OperatingSystemView : Gtk.Box {
+
+            var level_bar = new Gtk.LevelBar ();
+            level_bar.add_css_class (Granite.STYLE_CLASS_FLAT);
+            level_bar.add_css_class ("pink");
+
+            var details_box = new Gtk.Box (VERTICAL, 0);
+            details_box.append (target_label);
+            details_box.append (level_bar);
+
+            details_revealer = new Gtk.Revealer () {
+                child = details_box,
+                reveal_child = false
+            };
+
+            var link_image = new Gtk.Image.from_icon_name ("adw-external-link-symbolic");
+
+            var grid = new Gtk.Grid () {
+                valign = CENTER
+            };
+            grid.attach (image, 0, 0, 1, 2);
+            grid.attach (main_label, 1, 0);
+            grid.attach (details_revealer, 1, 1);
+            grid.attach (link_image, 2, 0, 2, 2);
+
+            child = grid;
+            add_css_class ("link");
+            get_goal_progress (level_bar, target_label);
+        }
+
+        private void get_goal_progress (Gtk.LevelBar levelbar, Gtk.Label target_label) {
+            var query = "{\"query\": \"query { organization(login: \\\"elementary\\\") { sponsorsListing { activeGoal { percentComplete, targetValue } } } }\"}";
+
+            var message = new Soup.Message ("POST", "https://api.github.com/graphql");
+            message.request_headers.append ("Authorization", "Bearer ghp_%s".printf (About.GITHUB_TOKEN));
+            message.request_headers.append ("Content-Type", "application/json");
+            message.request_headers.append ("User-Agent", "About.OperatingSystemView/1.0");
+            message.set_request_body_from_bytes (null, new Bytes (query.data));
+
+            var session = new Soup.Session ();
+            session.send_and_read_async.begin (message, GLib.Priority.DEFAULT, null , (obj, res) => {
+                try {
+                    var bytes = session.send_and_read_async.end (res);
+
+                    var output = (string) bytes.get_data ();
+                    if (output == null) {
+                        return;
+                    }
+
+                    var parser = new Json.Parser ();
+                    parser.load_from_data (output);
+
+                    var root = parser.get_root ();
+                    if (root.get_node_type () != OBJECT) {
+                        return;
+                    }
+
+                    var sponsors_listing = root.get_object ()
+                        .get_object_member ("data")
+                        .get_object_member ("organization")
+                        .get_object_member ("sponsorsListing")
+                        .get_object_member ("activeGoal");
+
+                    int64 percent_complete = sponsors_listing.get_int_member ("percentComplete");
+                    double target_value = sponsors_listing.get_double_member ("targetValue");
+
+                    levelbar.value = percent_complete / 100.0;
```

Comment:
You know what might be kind of fun is setting this with an Adw.Animation

Reply:
What do you think of this?
[Kooha-2024-12-06-07-18-17.webm](https://github.com/user-attachments/assets/57cdae62-2243-4278-9311-a8d2c49464ea)

Reply:
I did a branch to do it with Adw.Animation. It's a lot fewer lines and we can have a nice easing function https://github.com/elementary/switchboard-plug-about/pull/352